docs/section5/Rule5-39.md

```diff
@@ -1,32 +1,35 @@
-
 # Envelope - Rule 5-39  
-
+**Schema Version** 0.0.29
+**Primary Rule:** False  
 **Rule ID:** 5-39  
-**Rule Description:** It is acceptable to use either an annual average ground temperature or monthly average ground temperatures for calculation of heat loss through basement floors.  
-**Rule Assertion:** B-RMR ASHRAE229:weather.monthly_ground_temperature != NULL  
+**Rule Description:** It is acceptable to use either an annual average ground temperature or monthly average ground temperatures for calculation of heat loss through basement floors.
 **Appendix G Section:** Section 5 Envelope  
 **Appendix G Section Reference:** Section G3.1-14(b) Building Envelope Modeling Requirements for the Proposed design and Baseline  
 
-**Applicability:** All required data elements exist for B_RMR  
-**Applicability Checks:**  None  
-
-**Manual Check:** Yes  
-**Evaluation Context:** Each Data Element  
+**Applicability:** All required data elements exist for B_RMD
+**Applicability Checks:**  
+  1. B_RMD contains at least 1 floor surface adjacent to ground
+ 
+**Evaluation Context:** RMD 
 **Data Lookup:** None  
 **Function Call:** None  
 
 ## Rule Logic:  
+**Applicability Check 1**
+- If for-loop completes and no floor surfaces were found adjacent to ground, rule is not applicable: `is_applicable = false`
+- For each surface in the B_RMD: `for surface in B_RMD...Surface:` 
+  - If the surface is a floor and adjacent to ground, rule is applicable and continue to rule assertion: `if surface.classification == FLOOR and surface.adjacent_to == GROUND: is_applicable = true; break`
 
-**Rule Assertion:**  
-
-- Case 1: If B-RMR includes annual average or monthly average ground temperature: `if B_RMR.ASHRAE229.weather.monthly_ground_temperature: PASS`
-
-- Case 2: Else: `else: FAIL`
+**Rule Assertion:**
+- Case 1: If B_RMD has no floor surfaces adjacent to ground: `if not is_applicable: NOT_APPLICABLE`
+- Case 2: If RPD includes a ground temperature schedule: `if RPD.weather.ground_temperature_schedule: UNDETERMINED`
+- Case 3: Else: `else: FAIL`
 
 
 **Notes:**
-1. Do we need to check if RMR has basement floors and only apply this rule to models with basement floors?
-2. Update Rule ID from 5-52 to 5-39 on 10/26/2023
+1. Are slab-on-grade floors different from basement floors - Can we tell the difference from the schema?
```

Comment:
@JacksonJ-KC They are different.

slab-on-grade floor: that portion of a slab floor of the building envelope that is in contact with the ground and that is either above grade or is less than or equal to 24 in. below the final elevation of the nearest exterior grade. 

Below grade floors are unregulated in 90.1 as I understand it.

Based on the language of this rule in 90.1 it appears to only apply to below grade floors (although basement floor is not explicitly defined in 90.1 but seems reasonable to assume this means below grade floor). 

Can you check the elevation of the floor using the schema to determine whether it meets the definition of slab-on-grade versus below grade floor? I would think this would have to be possible since we need to check other rules related to slab on grade floors.

Reply:
If a zone has a below-grade wall surface would that then mean that the floor surface must be a basement floor? This could be a good solution! @weilixu @claperle

Reply:
It sounded like it needs to function such as `is_floor_basement_floor()`.

I do not think building segment can work. This data group includes many floors.
What could possibly work is to group all zones in a floor and then check if any of the walls are below grade wall, then all floor surface adjacent to ground in those zones are considered basement floor.

Reply:
I agree with @weilixu suggestion.

Reply:
Since we are only checking for applicability, if we find that a RMD has no below-grade wall surfaces then this means there are also no basement floor surfaces and the rule is not applicable. And since we agreed to add below-grade walls to also be included in this rule, then finding that a below-grade wall in an RMD means the rule is applicable if we are evaluating at the RMD level. I wrote the .md for a new function `get_basement_floors` to return a list of basement floor surface ids, but I don't believe we need it unless we evaluate at the surface level so I did not include in the PR.